### ISSUE TITLE
[BUG] RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by benjamincharity** [compodoc/compodoc#269]
##### **Overview of the issue**

Running the compodoc command throws a `RangeError`.

Command: `./node_modules/.bin/compodoc -p src/lib/tsconfig.compodoc.json -d docs --hideGenerator --toggleMenuItems all --theme stripe`

Full output from running command: https://pastebin.com/tKKK3yzu

Full repo can be viewed here: https://github.com/GetTerminus/terminus-ui/commits/log-in-form (note: I am generating docs for `src/lib`)

##### **Operating System, Node.js, npm, compodoc version(s)**

- macOS Sierra
- os: darwin x64
- node: 8.0.0
- compodoc 1.0.0-beta.13 **or** 1.0.0-beta.12
- angular 4.3.0

##### **Angular configuration, a `package.json` file in the root folder**

package.json also available here: https://github.com/GetTerminus/terminus-ui/blob/log-in-form/package.json

```
{
  "name": "@terminus/ui",
  "version": "0.0.0-development",
  "description": "Terminus UI Library",
  "main": "./bundles/ui.umd.js",
  "module": "./ui.es5.js",
  "es2015": "./ui.js",
  "typings": "./ui.d.ts",
  "files": [
    "dist"
  ],
  "author": "@terminus",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/GetTerminus/terminus-ui.git"
  },
  "engines": {
    "node": ">= 6.9.0",
    "npm": ">= 3.0.0"
  },
  "scripts": {
    "clean:lib": "rimraf out-tsc dist \"src/lib/**/!(systemjs-angular-loader).js\" \"src/lib/**/*.map\" \"src/lib/helpers.scss\"",
    "clean:styleguide": "rm -rf styleguide",
    "commitmsg": "validate-commit-msg",
    "cm": "./node_modules/.bin/git-cz",
    "predocs": "npm run markdown:test-links && npm run docs:toc",
    "docs": "npm run docs:ts && npm run docs:scss",
    "docs:scss": "node ./node_modules/.bin/nucleus --config config.nucleus.json",
    "docs:ts": "./node_modules/.bin/compodoc -p src/lib/tsconfig.compodoc.json -d docs --hideGenerator --toggleMenuItems all --theme stripe",
    "docs:toc": "doctoc --title '**Table of Contents**' --maxlevel 4 ./README.md",
    "prebuild": "npm run clean:lib",
    "build": "node tooling/build.js",
    "build:watch": "onchange 'src/lib/**/*.ts' 'src/lib/**/!(helpers).scss' -- npm run build",
    "build-demo": "cd src/demo && ng build --prod",
    "build-demo:watch": "cd src/demo && ng serve --port 4300",
    "build-scss": "node-sass src/lib -o src/lib --output-style expanded",
    "build-scss:watch": "node-sass src/lib -o src/lib --output-style expanded -w",
    "build-test": "tsc -p src/lib/tsconfig.spec.json",
    "build-test:watch": "tsc -p src/lib/tsconfig.spec.json -w",
    "preintegration": "npm run build && cd integration && npm run clean && npm install",
    "integration": "npm run integration:aot && npm run integration:jit",
    "integration:jit": "cd integration && npm run e2e",
    "integration:aot": "cd integration && npm run e2e:aot",
    "lint:ts": "tslint ./src/**/*.ts -t verbose",
    "lint:scss": "./node_modules/.bin/sass-lint -c .sass-lint.yml 'src/lib/**/*.scss' -v -q",
    "lint": "npm run lint:ts && npm run lint:scss",
    "markdown:test-links": "./node_modules/.bin/markdown-link-check ./README.md",
    "semantic-release": "semantic-release pre && npm publish && semantic-release post",
    "serve": "lite-server -c=tooling/bs-config.json --open false",
    "prestart": "npm run build && npm run build-demo && npm run build-scss",
    "start": "concurrently \"npm run build:watch\" \"npm run build-demo:watch\"",
    "pretest": "npm run build-test",
    "test": "concurrently \"npm run build-test:watch\" \"karma start karma.conf.js\"",
    "pretest:ci": "npm run build-test",
    "test:ci": "karma start karma.conf.js --single-run --code-coverage --sauceLabs",
    "update-scss-lib-extensions": "node tooling/replace-scss-lib-extension.js",
    "update-scss-demo-extensions": "node tooling/replace-scss-demo-extension.js",
    "watch-for-lib-css-extension": "onchange 'src/lib/**/*.js' -- npm run update-scss-lib-extensions"
  },
  "config": {
    "commitizen": {
      "path": "./node_modules/cz-customizable"
    },
    "cz-customizable": {
      "config": "./tooling/cz-config.js"
    }
  },
  "release": {
    "getLastRelease": "last-release-git",
    "verifyConditions": "condition-circle",
    "branch": "master"
  },
  "peerDependencies": {
    "@angular/animations": ">=4.0.0 <5.0.0 || >=4.0.0 <5.0.0",
    "@angular/core": ">=4.0.0 <5.0.0 || >=4.0.0 <5.0.0",
    "@angular/forms": ">=4.0.0 <5.0.0 || >=4.0.0 <5.0.0",
    "@angular/material": "2.0.0-beta.8",
    "@angular/router": ">=4.0.0 <5.0.0 || >=4.0.0 <5.0.0",
    "@angular/platform-browser": ">=4.0.0 <5.0.0 || >=4.0.0 <5.0.0",
    "angular2-ladda": "1.2.1"
  },
  "devDependencies": {
    "@angular/animations": "~4.3.0",
    "@angular/cli": "^1.1.3",
    "@angular/common": "~4.3.0",
    "@angular/compiler": "~4.3.0",
    "@angular/compiler-cli": "~4.3.0",
    "@angular/core": "~4.3.0",
    "@angular/flex-layout": "2.0.0-beta.8",
    "@angular/forms": "~4.3.0",
    "@angular/http": "~4.3.0",
    "@angular/material": "^2.0.0-beta.8",
    "@angular/platform-browser": "~4.3.0",
    "@angular/platform-browser-dynamic": "~4.3.0",
    "@angular/router": "~4.3.3",
    "@compodoc/compodoc": "^1.0.0-beta.13",
    "@types/jasmine": "2.5.53",
    "@types/node": "^8.0.1",
    "angular2-ladda": "^1.2.1",
    "camelcase": "^4.0.0",
    "codelyzer": "^3.0.1",
    "commitizen": "^2.9.6",
    "concurrently": "3.5.0",
    "condition-circle": "^1.5.0",
    "core-js": "^2.4.1",
    "cz-customizable": "^5.0.0",
    "doctoc": "^1.3.0",
    "execa": "^0.7.0",
    "glob": "^7.1.1",
    "hammerjs": "^2.0.8",
    "husky": "^0.14.0",
    "jasmine-core": "^2.5.2",
    "karma": "^1.5.0",
    "karma-chrome-launcher": "^2.0.0",
    "karma-cli": "^1.0.1",
    "karma-coverage": "^1.1.1",
    "karma-coverage-istanbul-reporter": "^1.3.0",
    "karma-html-reporter": "^0.2.7",
    "karma-jasmine": "^1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "karma-mocha-reporter": "^2.2.3",
    "karma-remap-coverage": "^0.1.4",
    "karma-sauce-launcher": "^1.1.0",
    "karma-sourcemap-loader": "^0.3.7",
    "karma-typescript": "^3.0.3",
    "last-release-git": "^0.0.3",
    "lite-server": "^2.2.2",
    "markdown-link-check": "^3.0.3",
    "merge-files": "^0.1.1",
    "node-sass": "^4.5.2",
    "nucleus-styleguide": "^1.1.1",
    "onchange": "^3.2.1",
    "remap-istanbul": "^0.9.5",
    "replace-in-file": "^2.5.0",
    "rimraf": "^2.6.1",
    "rollup": "^0.45.1",
    "rollup-plugin-sourcemaps": "^0.4.1",
    "rollup-plugin-uglify": "^2.0.1",
    "rxjs": "^5.0.1",
    "sass-lint": "^1.10.2",
    "semantic-release": "^6.3.6",
    "systemjs": "^0.20.14",
    "tslint": "^5.4.3",
    "typescript": "~2.4.2",
    "validate-commit-msg": "^2.12.2",
    "zone.js": "^0.8.4"
  },
  "dependencies": {
    "@angular/cdk": "^2.0.0-beta.8"
  }
}
```

##### **Compodoc installed globally or locally ?**

Locally. Run via  `npm run docs:ts` which runs ``./node_modules/.bin/compodoc -p src/lib/tsconfig.compodoc.json -d docs --hideGenerator --toggleMenuItems all --theme stripe``

##### **Motivation for or Use Case**

Docs are no longer generated :/

##### **Reproduce the error**

- Clone https://github.com/GetTerminus/terminus-ui/blob/log-in-form/
- Run `npm install`
- Run `npm run docs:ts`

##### **Related issues**

A few issues seem similar but I don't believe any were my exact issue.

- https://github.com/compodoc/compodoc/issues/263
- https://github.com/compodoc/compodoc/issues/217
- https://github.com/compodoc/compodoc/issues/209

##### **Suggest a Fix**

I have no clue :/ It was all working perfectly until this branch. Unfortunately this was a larger refactor along with a feature addition so it's not super easy to nail down which change caused this.

---

**Update:** I have tried excluding files that were being parsed just before the error occurred. This doesn't seem to help at all - the error still appears. The list of files in the `files` array when the error is thrown can be seen here: https://gist.github.com/benjamincharity/4836e94a01af64c87ec9798c8c3eaae6

My compodoc tsconfig:

```
{
  "compilerOptions": {
    "target": "ES5",
    "module": "commonjs",
    "moduleResolution": "node",
    "experimentalDecorators": true,
    "lib": ["es2015", "dom"]
  },
  "include": [
    "src/lib"
  ],
  "exclude": [
    "**/*.spec.ts",
    "**/*.mock.ts"
  ]
}
```

**Comment by benjamincharity** [comment]
Ok, this may not be as large of an issue... I have installed compodoc from the develop branch as suggested [here](https://github.com/compodoc/compodoc/issues/243#issuecomment-318885755) and this seemed to fix the issue! (and it is _much_ faster too 🎉 )

Feel free to leave open if this needs to be fixed for legacy support or close if this is fixed in the next release.

Thanks for an awesome tool!

**Comment by vogloblinsky** [comment]
Yes i confirm, fixed with 1.0.0-beta.14
